jsonify_stored/models/ir_exports.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class IrExport(models.Model):
+    """Because of jsonify.stored.mixin, models might depend on ir.exports records.
+       These records are identifiable by xmlid.
+       Therefore, when modifying a record we need to check that no models depend
+       on it, or update the corresponding model.
+    """
+
+    _inherit = "ir.exports"
+
+    def write(self, vals):
+        if "export_fields" in vals:
+            self._check_jsonify_stored()
+        res = super(IrExport, self).write(vals)
+        return res
+
+    def _check_jsonify_stored(self):
+        # if "jsonify.stored.mixin" not in self.env:
+        #     return  # might be necessary in some weird cases?
+        xml_ids = self.get_xml_id().values()
+        jsonify_mixin = self.env["jsonify.stored.mixin"]
+        jsonified_models = [self.env[m] for m in jsonify_mixin._inherit_children]
+        models = [m for m in jsonified_models if m._export_xmlid in xml_ids]
+        for model in models:
```

Comment:
are we sure this is not over killing when you have tons of records?
Maybe a manual action would be better?

Reply:
Do we really have so much jsonify model instances using jsonify_mixin? IMO not really. IMO this could stay as is.

Reply:
If you modify significantly the export lines (remove one, etc), all the exports have to be recomputed, so there's no way to go around it. But it should not happen much.

Reply:
> Do we really have so much jsonify model instances using jsonify_mixin? IMO not really. IMO this could stay as is.

You might have them. Eg: in the case of search engine bindings -> possibly a ton.
Also, one of the good reasons to make this split is to make this feature reusable across models/features. 
Meaning that potentially you can use it everywhere you want to expose data w. the same structure (eg: EDI, ESB, REST)

> If you modify significantly the export lines (remove one, etc), all the exports have to be recomputed, so there's no way to go around it. But it should not happen much.

Well, you might add a new field mapping on a db w/ more than 50000 records to recompute and the update will kick your face. As mentioned above: to me it's better if this becomes an aware decision/act.

If you want to do it in this place then you should do it atomically: recompute only what has changed (eg: remove field or compute only the new field). And this reminds me that we should find a better way to check if something has changed and find a way to recompute only what has changed. This is very hard to do because you might have nested information coming from nested related records.... hence let's discuss this for the next round :wink:

Reply:
@lmignon ping :)

Reply:
@len-foss Are we sure that the invalidation doesn't occur each time we update an addon defining ir.export.line into xml data files?

Reply:
We are sure that invalidation will occur n+1 time where is n is the number of lines at the export update. So that's bad.
The export should be put in noupdate, or put into a separate module that would only contain the export; otherwise, that would require really complicated mitigation strategies.